File: src/greedy/lc53.cc
#include <vector>

using namespace std;

class Solution {
 public:
  int maxSubArray(vector<int>& nums) {
    auto cur = nums[0];
    auto ans = nums[0];
    for (auto i = 1; i < nums.size(); i++) {
      cur = max(cur + nums[i], nums[i]);
      ans = max(ans, cur);
    }
    return ans;
  }
};

File: src/dp/lc53.py
from typing import List


class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        f = [0] * (len(nums) + 1)
        for i in range(len(nums)):
            f[i + 1] = max(f[i] + nums[i], nums[i])
        return max(f[1:])
